src/components/PeoplePage.tsx

```diff
@@ -1,33 +1,65 @@
 import { PeopleFilters } from './PeopleFilters';
-import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [visiblePeople, setVisiblePeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getPeople()
+      .then(peopleFromApi => {
+        setPeople(peopleFromApi);
+      })
+      .catch(() => setErrorMessage('Something went wrong'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const getParent = useCallback(
+    (name: string | null) => {
+      return people.find(parent => parent.name === name);
+    },
+    [people],
+  );
+
+  const peopleWithParents = useMemo(
+    () =>
+      people.map(person => {
+        const mother = getParent(person.motherName);
+        const father = getParent(person.fatherName);
+
+        return { ...person, mother, father };
+      }),
+    [getParent, people],
+  );
```

Comment:
You can simplify this like the following:

```suggestion
  const peopleWithParents = useMemo(
    () => {
        const getParent = (name: string | null) => {
           return people.find(parent => parent.name === name);
        }
        
        people.map(person => {
          const mother = getParent(person.motherName);
          const father = getParent(person.fatherName);

          return { ...person, mother, father };
        })
      },
    [getParent, people],
  );
```

Reply:
Or even better - move this to the `api`